lib/ecto/adapters/postgres/connection.ex

```diff
@@ -1015,6 +1015,11 @@ if Code.ensure_loaded?(Postgrex) do
       ["'\\x", Base.encode16(binary, case: :lower) | "'::bytea"]
     end
 
+    defp expr(%Ecto.Query.Tagged{value: bitstring, type: :binary}, _sources, _query)
+         when is_bitstring(bitstring) do
+      bitstring_literal(bitstring)
+    end
```

Comment:
> And then last thing is to add literal support in Ecto and then catch the tagged value here in the postgres adapter.

@greg-rychlewski not sure of this implementation.

In particular, I'm not sure if to leave `:binary` as type or not and I'm not sure if it is the best design to have a `bitstring_literal` function used both here and `default_type`

Reply:
oh so Jose agreed we should be handling it in Ecto so that the adapter sees it as `:bitstring`. So I think we can do something like in the post above this https://github.com/elixir-ecto/ecto/pull/4328#issuecomment-1950280536